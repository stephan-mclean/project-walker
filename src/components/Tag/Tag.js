import React from "react";
import PropTypes from "prop-types";
import styles from "./Tag.module.scss";

const Tag = ({ tag, children, className, ...otherProps }) => {
  const tagSquareClass = `${styles.tag} tag-${tag}-coloring margin-right`;
  return (
    <div
      className={`${
        styles.container
      } flex flex--center-y high-contrast-coloring ${className}`}
      {...otherProps}
    >
      <div className={tagSquareClass} />
      {children}
    </div>
  );
};

Tag.propTypes = {
  tag: PropTypes.oneOf([
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight"
  ]).isRequired,
  children: PropTypes.any,
  className: PropTypes.string
};

export default Tag;
