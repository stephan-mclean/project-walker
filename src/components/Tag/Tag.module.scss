.container {
  height: 2rem;

  .tag {
    width: 1rem;
    height: 100%;
  }
}
