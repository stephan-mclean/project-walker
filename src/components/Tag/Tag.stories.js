import React, { Fragment } from "react";
import { storiesOf } from "@storybook/react";

import Tag from "./Tag";

storiesOf("Components", module).add("Tag", () => {
  return (
    <Fragment>
      <Tag tag="one">one</Tag>
      <Tag tag="two">two</Tag>
      <Tag tag="three">three</Tag>
      <Tag tag="four">four</Tag>
      <Tag tag="five">five</Tag>
      <Tag tag="six">six</Tag>
      <Tag tag="seven">seven</Tag>
      <Tag tag="eight">eight</Tag>

      <Tag tag="one" className="transparent-high-contrast-coloring">
        transparent
      </Tag>
    </Fragment>
  );
});
