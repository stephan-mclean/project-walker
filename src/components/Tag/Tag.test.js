import React from "react";
import { shallow } from "enzyme";
import Tag from "./Tag";

describe("Testing Tag component", () => {
  it("Should render the correct content", () => {
    const tag = shallow(<Tag tag="one">one</Tag>);
    expect(tag).toMatchSnapshot();
    expect(tag.text()).toEqual("one");

    const tagWithChild = shallow(
      <Tag tag="two">
        <div>Child</div>
      </Tag>
    );
    expect(tagWithChild).toMatchSnapshot();
    expect(tagWithChild.contains(<div>Child</div>)).toBeTruthy();
  });

  it("Should respond to the users click", () => {
    const onClick = jest.fn();
    const tag = shallow(
      <Tag tag="one" onClick={onClick}>
        one
      </Tag>
    );

    tag.simulate("click");

    expect(onClick).toHaveBeenCalled();
  });

  it("Should allow using a custom class", () => {
    const tag = shallow(
      <Tag tag="one" className="primary-coloring">
        Tag
      </Tag>
    );
    expect(tag.hasClass("primary-coloring")).toEqual(true);
  });

  it("Should render the correct tag square color", () => {
    const tag = shallow(<Tag tag="one">Tag</Tag>);
    expect(tag.find(".tag-one-coloring").exists()).toBeTruthy();
  });
});
