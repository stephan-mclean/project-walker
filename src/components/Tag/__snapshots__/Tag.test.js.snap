// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Tag component Should render the correct content 1`] = `
<div
  className="container flex flex--center-y high-contrast-coloring undefined"
>
  <div
    className="tag tag-one-coloring margin-right"
  />
  one
</div>
`;

exports[`Testing Tag component Should render the correct content 2`] = `
<div
  className="container flex flex--center-y high-contrast-coloring undefined"
>
  <div
    className="tag tag-two-coloring margin-right"
  />
  <div>
    Child
  </div>
</div>
`;
